a10_octavia/controller/worker/tasks/nat_pool_tasks.py

```diff
@@ -76,24 +76,25 @@ def execute(self, loadbalancer, vthunder, flavor_data=None):
 
     @axapi_client_decorator
     def revert(self, loadbalancer, vthunder, flavor_data=None, *args, **kwargs):
-        natpool_flavor_list = flavor_data.get('nat_pool_list')
-        natpool_flavor = flavor_data.get('nat_pool')
-        try:
-            if natpool_flavor_list:
-                for nat_pool in natpool_flavor_list:
-                    pool_name = nat_pool['pool_name']
+        if flavor_data:
+            natpool_flavor_list = flavor_data.get('nat_pool_list')
+            natpool_flavor = flavor_data.get('nat_pool')
+            try:
+                if natpool_flavor_list:
+                    for nat_pool in natpool_flavor_list:
+                        pool_name = nat_pool['pool_name']
+                        LOG.warning("Reverting creation of nat-pool: %s", pool_name)
+                        self.axapi_client.nat.pool.delete(pool_name)
+                if natpool_flavor:
+                    pool_name = natpool_flavor['pool_name']
```

Comment:
same here?

Reply:
Yes....modifying it as `pool_name = natpool_flavor.get('pool_name')`